### ISSUE TITLE
Newly Created Yes/No Parameters Default Value Checked On

### ISSUE THREAD

**Issue by EATREVITPOOPCAD** [erfajo/OrchidForDynamo#247]
### Operating system
Windows 10

### Revit and Dynamo version
Revit: 2023.1.3
Dynamo: 2.16.2
Orchid: 2.16.5.6225

### What did you do? 
Created a Yes/No parameter

### What did you expect to see?
Newly created Yes/No parameters checked off.

### What did you see instead?
Newly created Yes/No parameters come in checked on.

Although this is a default behavior in Revit when you create a Yes/No parameter, I believe for batch creation of parameters a false value is a safer bet. (similar to how newly created length parameters are 0, having them anything than that has higher chances of mistakes slipping through the cracks)

PS: I am pretty sure the node returns all shared parameters that are being added, even if that parameter already existed before, in which case it was technically never added. This makes it difficult for me in dynamo to isolate those newly added Yes/No parameters and set their value to false.

**Comment by erfajo** [comment]
Do you have a graph file? you can add .txt to the extension, and then you can upload the file.

however, are you aware of the distinction between the node “create” and “add”? the first will force creation no matter what, and the latter will only add new and leave existing as is, however, updating the provided categories!

I will try testing myself for Yes/No parameter, however, I do not add any “default” values to new parameters, so if it does change this, then it is a bug… but maybe is the bug a Revit API bug and not a bug in my node. I will return on this.

**Comment by EATREVITPOOPCAD** [comment]
[Add-Shared-Param_BAD-LOGGING.txt](https://github.com/user-attachments/files/16038794/Add-Shared-Param_BAD-LOGGING.txt)

Couldn't get back to this for a while, but better late than ever :D

I was not aware of the difference between **create** & **add**, good to know. **Add** is the node I need. I attached the dynamo file. It works for parameter creation, except I was not able to get logging CSV functionality to show which families a parameter was added to, and which families already had it. So, the CSV logging is essentially useless, I just left it in case I can get it to work one day.

Regarding the **Yes/No** parameter initial value, there is no glitch. It is Revit default behavior, I was just ranting that if a **Yes/No** parameter is created, a default value of **False** is better than Revit's current assignment of the **True** value to newly created **Yes/No** parameters. But I can see how someone could argue otherwise... Either way, can be resolved with another post-run Dynamo script to set those values as needed.

**Comment by erfajo** [comment]
Thanks, I will try to see what I can do. However, could I also have a snip of the other files?
the excel file with parameters, one family, so I can see what kind of families it is. Maybe it could also be an idea to have the csv file!?

and no need for the directory contents node I have a node for that as well :-)
![image](https://github.com/erfajo/OrchidForDynamo/assets/2844374/fc83e0f9-e9cf-4727-bd35-00b77438e2a0)

**Comment by erfajo** [comment]
I have been looking at your issue and may say, that this is the default behavior from the Revit API, to set the parameter as checked. I can't change that behavior, however, it would be doable to add a procedure to set the parameter in such cases as unchecked. But after evaluating the pros and cons I have decided that if the Revit team thinks “yes” is better to set than “No”, then is this also going to be my default value.

I have added som testing in your graph, if you want to see that :-)
[Add-Shared-Param_BAD-LOGGING_erfajo.txt](https://github.com/user-attachments/files/16232626/Add-Shared-Param_BAD-LOGGING_erfajo.txt)
